text/0117-typo-tolerance-settings-api.md

```diff
@@ -0,0 +1,202 @@
+
+## 2. Motivation
+
+Tolerance to typos is critical when building an enjoyable search experience. Meilisearch ranks typo-free matched documents above others based on the default settings that we believe are best for most users.
+
+Meilisearch can adapt to different use-cases by providing customization options for the typo tolerance feature.
+
+## 3. Functional Specification
+
+### 3.1. `typoTolerance` API resource definition
+
+| Field                                            | Type            | Required |
+|--------------------------------------------------|-----------------|----------|
+| [enabled](#311-enabled)                          | Boolean         | False    |
+| [disableOnAttributes](#312-disableonattributes)  | Array of String | False    |
+| [disableOnWords](#313-disableonwords)            | Array of String | False    |
+| [minWordSizeFor1Typo](#314-minwordsizefor1typo)  | Integer         | False    |
+| [minWordSizeFor2Typos](#315-minwordsizefor2typos)| Integer         | False    |
+
+#### 3.1.1. `enabled`
+
+- Type: Boolean
+- Required: False
+- Default: `true`
+
+Whether the typo tolerance feature is enabled.
+
+##### 3.1.1.1. Impacts on the `typo` ranking rule
+
+The presence of `typo` in the ranking rules setting does not influence the activation/deactivation of the typo tolerance feature.
+
+If the `rankingRules` parameter of the index settings does not contain the `typo` rule, the results are not sorted according to the number of typos found.
+
+The typo tolerance feature is still used to match documents.
+
+### 3.1.2. `disableOnAttributes`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disableOnAttributes` disable the typo tolerance feature on the specified attributes set.
+
+Any attributes defined in `disableOnAttributes` won't have their values matched by the typo tolerance.
+
+### 3.1.3. `disableOnWords`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disableOnWords` disable the typo tolerance feature for a set of search query terms.
+
+#### 3.1.3.4. Example
+
+If `Javascript` is specified in `disableOnWords`, the engine won't apply the typo tolerance on the query term `Javascript` if typed at search time.
+
+### 3.1.4. `minWordSizeFor1Typo`
+
+- Type: Integer
+- Required: False
+- Default: `5`
+
+`minWordSizeFor1Typo` customize the minimum size for a word to tolerate 1 typo.
+
+> Given the default value `5`, the search engine starts to apply typo tolerance on a query term if its length is at least equal to 5 characters.
+
+> See [2.1. Typos calculation section](#21-typos-calculation)
+
+### 3.1.5. `minWordSizeFor2Typos`
+
+- Type: Integer
+- Required: False
+- Default: `9`
+
+`minWordSizeFor2Typos` customize the minimum size for a word to tolerate 2 typos.
+
+> Given the default value `9`, the search engine handles up to 2 typos on a query term if its length is at least equal to 9 characters.
+
+> See [2.1. Typos calculation section](#21-typos-calculation)
```

Comment:
How are `3` and `4` typos impacted by that?
Lets say I set `minWordSizeFor2Typos` to `10`, will it set `10` for `3` and more typos as well?

Reply:
Meilisearch does not tolerate typos `>2` 🤓 It is not planned that Meilisearch will tolerate more typos but this may be a future possibility.

Let me know if you need more details about that!

Reply:
I understand! Thanks :)